luarules/gadgets/game_unit_sharing_mode.lua

```diff
@@ -0,0 +1,109 @@
+local gadget = gadget ---@type Gadget
```

Comment:
belongs factored into team transfer. I cherry-picked this recently into this branch so ensure the logic is merged correctly to account for the dropdown